lib/buffer.js

```diff
@@ -42,34 +42,92 @@ function alignPool() {
   }
 }
 
-
-function Buffer(arg, encoding) {
+/**
+ * The Buffer() construtor is "soft deprecated" ... that is, it is deprecated
+ * in the documentation and should not be used moving forward. Rather,
+ * developers should use one of the three new factory APIs: Buffer.from(),
+ * Buffer.allocUnsafe() or Buffer.alloc() based on their specific needs. There
+ * is no hard deprecation because of the extent to which the Buffer constructor
+ * is used in the ecosystem currently -- a hard deprecation would introduce too
+ * much breakage at this time. Eventually, however, we should re-evaluate that
+ * choice.
```

Comment:
Even if it's "hard deprecated" it can never be removed. Since it still needs to act as a constructor for any typed array methods, or anything inheriting from Buffer.